werkzeug/utils.py

```diff
@@ -423,13 +423,7 @@ def import_string(import_name, silent=False):
             return sys.modules[import_name]
 
         module_name, obj_name = import_name.rsplit('.', 1)
-        try:
-            module = __import__(module_name, None, None, [obj_name])
-        except ImportError:
-            # support importing modules not yet set up by the parent module
-            # (or package for that matter)
-            module = import_string(module_name)
```

Comment:
This also made me raise an eyebrow, but I could not understand what "not yet set up" implied either. However, the case that I introduced in the tests, does cause the `ImportError` of line 428 to be raised.

FWIW I use this patch with https://github.com/inveniosoftware/flask-registry and had no obvious issues.